code/modules/reagents/oldchem/reagents/reagents_food.dm

```diff
@@ -156,6 +156,11 @@
 	color = "#B1B0B0"
 	overdose_threshold = 100
 
+/datum/reagent/sodiumchloride/reaction_mob(mob/living/M, method=TOUCH, reac_volume)
+	if(istype(M) && M.has_bane(BANE_SALT))
```

Comment:
no need to apply istype checks for living; `reaction_mob` is never called on `mob` subtypes.

also, it should be `/datum/reagent/sodiumchloride/reaction_mob(mob/living/M, method=TOUCH, volume)`